### PR TITLE
fix: allow a wider range of filenames

### PR DESCRIPTION
**Pull request by lajp**
This PR improves on filename matching in order to e.g. render .jpeg files as parts of the pdf.
Additionally it disallows unsupported filenames entirely, returing an error if encountered.

This includes the current progress of #23 but is to be merged beforehand in order to get the hotter fixes in ASAP

### CODE REVIEW

**Comment by lajp** [comment]
This requires the addition of the env variable `ALLOWED_ORIGINS=http://localhost:3000,https://tietokilta.fi`
and the `EXPOSE` env var is changed to `BIND_ADDR=0.0.0.0`

File: src/error.rs [hunk 1]
```diff
@@ -16,6 +16,8 @@ pub enum Error {
     MultipartRejection(#[from] axum::extract::multipart::MultipartRejection),
     #[error("Missing filename multipart")]
     MissingFilename,
+    #[error("Unsupported file format: {0}")]
```
Comment (kahlstrm) [inline src/error.rs]: I wonder if we should provide a list of supported file formats. Not 100% necessary.

File: src/main.rs [hunk 2]
```diff
@@ -16,6 +18,43 @@ mod tests;
...
+#[command(version, about, long_about = None)]
+struct LaskugenConfig {
+    #[clap(flatten)]
+    mailgun: MailgunConfig,
+    /// The listen port for the HTTP server
+    #[clap(long, env, required = false, default_value = "3000")]
+    port: u16,
+    /// The ip address to bound by the HTTP server
+    #[clap(long, env, required = false, default_value = "127.0.0.1")]
+    bind_addr: std::net::IpAddr,
```
Comment (kahlstrm) [inline src/main.rs:50]: I guess this could be set to 0.0.0.0 in the `Dockerfile`, as that is probably the default you want to use with containers.

**Review by kahlstrm — APPROVED** [review]
Looks good, just need to add the default host setting to the Dockerfile, otherwise this will not work without infra changes.

File: src/error.rs (hunk 1 above)
Comment (lajp) [reply src/error.rs]: Yeah, could be. Also the frontend should refuse to send unsupported files

**Pull request merged**